yjit_codegen.c

```diff
@@ -4513,6 +4513,25 @@ gen_toregexp(jitstate_t *jit, ctx_t *ctx, codeblock_t *cb)
     return YJIT_KEEP_COMPILING;
 }
 
+static codegen_status_t
+gen_intern(jitstate_t *jit, ctx_t *ctx, codeblock_t *cb)
+{
+    // Save the PC and SP because we might allocate
+    jit_prepare_routine_call(jit, ctx, REG0);
+
+    x86opnd_t str = ctx_stack_pop(ctx, 1);
+
+    mov(cb, C_ARG_REGS[0], str);
+
+    call_ptr(cb, REG0, (void *)&rb_str_intern);
+
+    // Push the return value
+    x86opnd_t stack_ret = ctx_stack_push(ctx, TYPE_UNKNOWN);
+    mov(cb, stack_ret, RAX);
```

Comment:
I guess we know that it's a symbol, but we don't know whether it's an immediate or an on heap one. I like `TYPE_UNKNOWN` anyways, though. Lower risk and I don't think currently knowing the class without the handle shape yields better code anywhere.